src/portable/portablectl.c

```diff
@@ -258,15 +255,12 @@ static int get_image_metadata(sd_bus *bus, const char *image, char **matches, sd
         _cleanup_(sd_bus_message_unrefp) sd_bus_message *m = NULL;
         _cleanup_(sd_bus_error_free) sd_bus_error error = SD_BUS_ERROR_NULL;
         uint64_t flags = arg_force ? PORTABLE_FORCE_SYSEXT : 0;
-        const char *method;
         int r;
 
         assert(bus);
         assert(reply);
 
-        method = strv_isempty(arg_extension_images) ? "GetImageMetadata" : "GetImageMetadataWithExtensions";
```

Comment:
wouldn't it be nicer to keep using the old function bug just conditoinalize this on `arg_force` as well?

i mean, usually we try to be conservative with what the client side tools call, to retain some compat with network transparency via "portablectl -H …" or container transparency with "portablectl -M …". i.e. the rule is to not call new functions needlessly if the old functions do too. And afaics if arg_force is not used the old function should be entirely fine, no?

Reply:
updated